pyomo/solvers/plugins/solvers/MIPCL.py

```diff
@@ -0,0 +1,119 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import logging
+
+import pyutilib.services as services
+
+from pyutilib.misc import Bunch, Options
+from pyutilib.services.TempfileManager import create_tempfile
+
+from pyomo.util.plugin import alias
+from pyomo.opt import SolverResults
+from pyomo.opt.base import OptSolver, SolverFactory, ProblemFormat, ResultsFormat
+from pyomo.opt.solver import SystemCallSolver
+from pyomo.opt.base.solvers import _extract_version
+
+from six import iteritems, string_types
+
+logger = logging.getLogger('pyomo.solvers')
+
+_mipcl_version = None
+
+class MIPCL(OptSolver):
+    """The MIPCL LP/MIP solver"""
+
+    alias('mipcl', doc='The MIPCL LP/MIP solver')
+
+    def __new__(cls, *args, **kwds):
+        try:
+            mode = kwds['solver_io']
+            if mode is None:
+                mode = 'mps'
+            del kwds['solver_io']
+        except KeyError:
+            mode = 'mps'
+        if mode == 'mps':
+            opt = SolverFactory('_mipcl_shell', **kwds)
+            opt.set_problem_format(ProblemFormat.mps)
+            return opt
+        if mode == 'os':
+            opt = SolverFactory('_ossolver', **kwds)
+        else:
+            logger.error('Unknown IO type: %s' % mode)
+            return
+        opt.set_options('solver=mps_mipcl')
+        return opt
+
+class MIPCLSHELL(SystemCallSolver):
+    """Shell interface to the MIPCL LP/MIP solver"""
+
+    alias('_mipcl_shell', doc='Shell interface to the MIPCL LP/MIP solver')
+
+    def __init__(self, **kwds):
+        kwds['type'] = 'mipcl'
+        SystemCallSolver.__init__(self, **kwds)
+
+        self._valid_problem_formats = [ProblemFormat.mps]
+        self._valid_result_formats = {ProblemFormat.mps: ResultsFormat.sol}
```

Comment:
Can you help me understand this debugger output.  I have a little test script that calls the 'mipcl' solver and it returns a nonzero error.  It seems like _status = self._apply_solver() correctly calls the executable but  _status.log returns the console output or std_out, instead of the .sol file the executable generates, also _status.rc is 1. 

> --> 589             _status = self._apply_solver()
    590             if hasattr(self, '_transformation_data'):
    591                 del self._transformation_data

> c:\miniconda3\envs\opt\lib\site-packages\pyomo\opt\base\solvers.py(590)solve()
    588 
    589             _status = self._apply_solver()
--> 590             if hasattr(self, '_transformation_data'):
    591                 del self._transformation_data
    592             if not hasattr(_status, 'rc'):


>ipdb> > c:\miniconda3\envs\opt\lib\site-packages\pyomo\opt\base\solvers.py(592)solve()
    590             if hasattr(self, '_transformation_data'):
    591                 del self._transformation_data
--> 592             if not hasattr(_status, 'rc'):
    593                 logger.warning(
    594                     "Solver (%s) did not return a solver status code.\n"

> c:\miniconda3\envs\opt\lib\site-packages\pyomo\opt\base\solvers.py(597)solve()
    595                     "This is indicative of an internal solver plugin error.\n"
    596                     "Please report this to the Pyomo developers." )
--> 597             elif _status.rc:
    598                 logger.error(
    599                     "Solver (%s) returned non-zero return code (%s)"

> c:\miniconda3\envs\opt\lib\site-packages\pyomo\opt\base\solvers.py(598)solve()
    596                     "Please report this to the Pyomo developers." )
    597             elif _status.rc:
--> 598                 logger.error(
    599                     "Solver (%s) returned non-zero return code (%s)"
    600                     % (self.name, _status.rc,))


>ipdb> _status
{'rc': 1, 'log': 'Usage: mps_pMIPCL/mps_MIPCL [options] filename.mps[.gz]\n   -solfile {file name for storing solution}   -threads {number of thread}\n   -time {time in seconds}\n   -gap {duality gap in percents}\n   -objBound {lover or upper bound on the objective}\n   -LPmethod {PRIME, DUAL}\n   -pricingRule {DANCIG,STEEPEST,PARTIAL}\n   -sepRule {DANCIG,STEEPEST,PARTIAL}\n   -preprocess {0,1}\n   -cuts l,h (l,h in {-1,0,1,...} are #cuts for, resp., low and high nodes)\n   -click {cut pattern}\n   -knapsack {cut pattern}\n   -gomory {cut pattern}\n   -mod2 {cut pattern}\n   -mir {cut pattern}\n           file_name\n----------------------------------------------------------------\ncut pattern depth1,mod1,depth2,mod2  where depth1,depth2 in {-1,0,1,2,...\n'}